modules/parachains/src/lib.rs

```diff
@@ -467,28 +576,21 @@ pub mod pallet {
 				Error::<T, I>::HeaderChainStorageProof(HeaderChainError::StorageProof(e))
 			})?;
 
-			Ok(PostDispatchInfo { actual_weight: Some(actual_weight), pays_fee: Pays::Yes })
-		}
-
-		/// Change `PalletOwner`.
-		///
-		/// May only be called either by root, or by `PalletOwner`.
-		#[pallet::call_index(1)]
-		#[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
-		pub fn set_owner(origin: OriginFor<T>, new_owner: Option<T::AccountId>) -> DispatchResult {
-			<Self as OwnedBridgeModule<_>>::set_owner(origin, new_owner)
-		}
+			// check if we allow this submission for free
+			let is_free = total_parachains == 1
+				&& free_parachain_heads == total_parachains
+				&& SubmitFinalityProofHelper::<T, T::BridgesGrandpaPalletInstance>::can_import_anything_for_free();
```

Comment:
Not sure if we should use the same `pallet_bridge_grandpa::FreeHeadersRemaining` value both for Grandpa and Parachains. Would it make sense to have a separate one for parachains ?

Reply:
Let's go with same as Grandpa for now

Reply:
The idea with those free headers, when we allow one free parachain header per free relay chain header, both pallets are acting as a whole. They've already been a part of a "single" bridge, but now they're even closer one to another. So we may use single storage + config param for that. In the end, this config param protects the block capacity, which is shared by all pallets.

Adding more storage entries/constants would just make it harder to configure IMO